### DISCUSSION TITLE
2024-07-22 회의록

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#65] [회의록]
## 필요한 유틸 함수(utils) / hooks

- timestamptz 포매팅 함수
  - 2023-03-12 00:17:00+00 -> 23.03.12
- getUser()
  - header에서 호출 -> global state??
- 함수를 만들 때 공용으로 쓰일 것 같은지 고민해보기 -> 쓰일 것 같다면 만들고 내용 슬랙에 공유하기

## 오늘 일정 안내

- 09:10 ~ 오전 회의
- 14:00 ~ 1차 회의
- 19:00 ~ 2차 회의
- 시간 조정 필요해보임 -> 다른 조 보니 오전까지 3번 회의를 하는데, 아침에 회의까진 아니어도 어떤 내용 진행할 지 정도는 인사 나누거나 공유하고 시작해야 할 필요성 ~~
- 12:20~13:00 : 오은 모의면접

## 깃허브 사용법

- discussion PR convention 스니펫
- projects 세부 티켓 입력 방법, startdate, enddate -> 타임라인 보기, 이슈로 넘기기, 담당자 지정하기, 순서 DnD로 바꾸기

## supabase 설명

- 권한이 아직 안 열려있을 것 -> 바로 이것부터 열게요

## 프로젝트 구조 설명

```
.eslintrc.js
.gitignore
.vscode
extensions.json
settings.json
README.md
next.config.mjs
package.json
postcss.config.mjs
public
manifest.json
next.svg
sw.js 🍀 build 시 인코딩 되면 디스커션 스니펫에서 복붙해서 복구!
test_icon.png
vercel.svg
src
api-services
│  │  └─ temp.ts
│  ├─ app
(providers)
│  │  │  ├─ (authenticated) 🍀 인증된 사용자만 접근 가능한 라우트
│  │  │  │  └─ layout.tsx
│  │  │  ├─ (conditional) 🍀 인가/비인가 회원 둘 다 조건부로 보여주는 라우트
│  │  │  │  └─ layout.tsx
│  │  │  ├─ (public) 🍀 비인가 회원에게도 보여주는 라우트
│  │  │  │  └─ page.tsx
layout.tsx
│  │  ├─ favicon.ico
│  │  ├─ globals.css
│  │  └─ layout.tsx
components
│  │  ├─ atoms ⭐️ 원자 : 더 이상 분해할 수 없는 컴포넌트
│  │  │  └─ temporary.ts
│  │  ├─ molecules ⭐️ 분자 : 두 가지 이상의 컴포넌트가 결합된 컴포넌트
│  │  │  ├─ Footer.tsx
│  │  │  └─ Header.tsx
│  │  ├─ organism ⭐️ 유기체 : 두 가지 이상의 molecules(분자)가 결합된 컴포넌트
│  │     └─ pwa
│  │        └─ InstallPromptHandler.tsx
│  ├─ constants 🍀 상수 / 쿼리키 모아 놓는 곳
│  │  ├─ common.constants.ts ⭐️ 일반 상수 모아 놓는 곳
│  │  └─ query.constants.ts ⭐️ 쿼리 키 모아놓는 곳
│  ├─ contexts
│  │  └─ temp.ts
│  ├─ data
│  │  └─ location.ts
hooks 🍀 react에서만 사용할 수 있는 hook / client component에서만 사용 가능
│  │  └─ useCheckPwa.ts
│  ├─ middleware.ts
│  ├─ providers
│  │  └─ QueryProvider.tsx
│  ├─ types 🍀 타입 모아 놓는 곳
│  │  └─ supabase.ts
│  └─ utils
│     ├─ pwa
│     │  └─ isPWA.ts
│     └─ supabase
│        ├─ client.ts
│        ├─ middleware.ts
│        └─ server.ts
├─ supabase
│  ├─ .gitignore
│  ├─ config.toml
│  └─ seed.sql
├─ tailwind.config.ts
├─ tsconfig.json
└─ yarn.lock
```

- 은님 집도

## 와이어프레임

- 와이어프레임에서 부족하거나 공수가 많을 것 같은 요소 캐치해서 다음 회의 때 자세하게 말씀해주세요

## 설치 필요한 라이브러리

- zustand
- alert 라이브러리 X -> 커스텀 UI O
  - 외부 라이브러리 의존도 낮추기
- axios X -> Next.js 내장 메서드 O
  - Next,js의 내장 메서드로 충분히 사용 가능한 만큼 라이브러리 의존도 낮추기

## 이번 주 개인별 페이지 제작 일정과 마감기한

| 페이지     | 담당자 | UI 제작 시작 | UI 제작 종료 |
|---------|-----|----------|----------|
| 여정상세페이지 | 오은  |   7/27   |   7/29    |
| 여정목록페이지 | 오은  |   7/27   |   7/29    |
| 로그인     | 오은  |    7/22     |   7/24   |
| 회원가입    | 오은  |   7/22     |   7/24   |
| 온보딩     | 오은  |   7/27   |   7/29    |
| 스토리작성   | 오은  |   7/24     |  7/26   |
| 스토리목록   | 오은  |   7/24     |   7/26   |
| 스토리상세   | 오은  |   7/24    |   7/26   |
| 여정작성페이지 | 김병준 |    7/22     |   7/23        |
| 여정수정페이지 | 김병준 |    7/22      |   7/23       |
| 여정전체보기페이지 | 김병준 |    7/23      |    7/24      |
| 버디프로필   | 김병준 |     7/24     |    7/25      |
| 버디프로필변경 | 김병준 |   7/24       |   7/25       |
| 인기버디즈목록 | 김병준 |     7/25     |   7/26       |
| 검색페이지   | 곽현정 |    7/22      |   7/24       |
| 검색결과페이지 | 곽현정 |   7/24       |   7/25       |
| 메인      | 김민지 | 7/22 | 7/24 |
| 채팅목록페이지 | 김민지 | 7/23 | 7/25 |
| 채팅상세페이지 | 김민지 | 7/23 | 7/25 |
| 채팅요청모달  | 김민지 | 7/24 | 7/26 |
| 튜토리얼    | 최유상 |          |          |
| 결제페이지   | 최유상 |          |          |

- 위 내용은 페이지 단위의 UI만 나와 있고, 누락된 부분 있는지 각 담당자가 체크해서 월요일 저녁 회의에서 논의 (이강민 튜터님 피드백 사항)
- 각 페이지의 전체 단위가 아니라, 기능별, 컴포넌트별로 잘게 티켓을 나누어 마감 기한을 회의 때 말씀해주세요
  - OO 페이지 제작 (~00/00일까지 초안 완료)
    - UI 제작 : 00/00일 시작 ~ 00/00일 종료
    - OO 컴포넌트 : 00/00일 시작 ~ 00/00일 종료
    - XX 컴포넌트 : 00/00일 시작 ~ 00/00일 종료

## 디자이너 미팅

- 홈 화면 '추천 인기 버디즈' 변경 사항
  - '여행 중인 지역' - > 이름/나이
  - 이름/나이 -> 온도지수

## 일일 2차 회의 (14:00~15:30)

- Projects 타임라인 화면 공유하며 일정 공유
- 개인별 티켓 세부적으로 나누기
  - 커스텀 alert -> 오은 제작
  - 헤더 -> 김민지 제작
- 반응형 레이아웃 (provider)에서 적용하고 시작하기 -> 오은 셋업

**Comment by eunohhh** [comment]
## 이번 주 개인별 페이지 제작 일정과 마감기한(오은)

| 페이지     | 담당자 | UI 제작 시작 | UI 제작 종료 |
|---------|-----|----------|----------|
| 여정상세페이지 | 오은  |   7/27   |   7/29    |
| 여정목록페이지 | 오은  |   7/27   |   7/29    |
| 로그인     | 오은  |    7/22     |   7/24   |
| 회원가입    | 오은  |   7/22     |   7/24   |
| 온보딩     | 오은  |   7/27   |   7/29    |
| 스토리작성   | 오은  |   7/24     |  7/26   |
| 스토리목록   | 오은  |   7/24     |   7/26   |
| 스토리상세   | 오은  |   7/24    |   7/26   |

**Comment by slsl2** [comment]
## 이번 주 개인별 페이지 제작 일정과 마감기한

| 페이지     | 담당자 | UI 제작 시작 | UI 제작 종료 |
|---------|-----|----------|----------|
| 메인      | 김민지 | 7/22 | 7/24 |
| 채팅목록페이지 | 김민지 | 7/23 | 7/25 |
| 채팅상세페이지 | 김민지 | 7/23 | 7/25 |
| 채팅요청모달  | 김민지 | 7/24 | 7/26 |

**Comment by hyunjeongkwak** [comment]
|페이지|담당자|UI 제작 시작|UI 제작 종료|
|:------:|---|:---:|:---:|
|검색 메인 페이지|곽현정|7/22|7/23|
|위치 검색 페이지|곽현정|7/23|7/24|
|날짜 선택 페이지|곽현정|7/24|7/24|
|검색 결과 페이지|곽현정|7/24|7/25|
